hpx/util/pool_timer.hpp

```diff
@@ -0,0 +1,114 @@
+//  Copyright (c) 2016 Bibek Wagle
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_UTIL_POOL_TIMER)
+#define HPX_UTIL_POOL_TIMER
+
+#include <hpx/config.hpp>
+#include <hpx/util/function.hpp>
+#include <hpx/lcos/local/spinlock.hpp>
+
+#include <boost/thread/locks.hpp>
+#include <boost/shared_ptr.hpp>
+#include <boost/enable_shared_from_this.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
```

Comment:
Could you please try to minimize the necessary #includes? You probably do not need to include all of Asio here just to declare the deadline_timer.